src/Microsoft.Diagnostics.WebSocketServer/WebSocketServerImpl.cs

```diff
@@ -172,24 +172,24 @@ private static void ParseWebSocketURL(string endPoint, out Uri uri)
             uriToParse = endPoint;
         }
 
-        string[] supportedSchemes = new string[] { "ws", "wss", "http", "https" };
 
         if (!string.IsNullOrEmpty(uriToParse) && Uri.TryCreate(uriToParse, UriKind.Absolute, out uri))
         {
-            bool supported = false;
-            foreach (string scheme in supportedSchemes)
+            if (uri.Scheme == "ws")
```

Comment:
Does this just change the Scheme from ws to http and from wss to https? If that is the case, can we change the logic for how the Scheme is used to set the option  https://github.com/dotnet/diagnostics/blob/d2f05caf4a97d8dc2a75d910d5d0c1170e2ed640/src/Microsoft.Diagnostics.WebSocketServer/WebSocketServerImpl.cs#L45, so we don't duplicate the work of building the Uri that is done in `Uri.TryCreate`.

Curious, is there something else we can change to make ws + wss start a server, or will it not work with that prefix?

Reply:
@mdh1418 I don't understand want you mean, could you please elaborate ?

The use of `ParseWebSocketURL` is just few lines above and `EmbeddedWebSocketServer.Options` is not `Uri`

Reply:
done, much nicer, thanks